packages/faustwp-cli/src/healthCheck/verifyGraphQLEndpoint.ts

```diff
@@ -0,0 +1,44 @@
+import 'isomorphic-fetch';
+
+import { infoLog, errorLog } from '../stdout/index.js';
+import { getGraphqlEndpoint } from '../utils/index.js';
+
+/**
+ * Checks if the WordPress GraphQL endpoint is available to use.
+ */
+export async function verifyGraphQLEndpoint() {
+  const graphqlEndpoint = getGraphqlEndpoint();
+
+  try {
+    // Perform GraphQL request.
+    const response: Response = await fetch(graphqlEndpoint, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ query: '{ __typename }' }),
+    });
+
+    // Expect a valid GraphQL response.
+    const json: {
+      data: {
+        __typename: 'string';
+      };
+    } = await response.json();
+
+    // eslint-disable-next-line no-underscore-dangle
+    if (json.data.__typename) {
+      infoLog('Discovered WPGraphQL endpoint!');
+      return true;
+    }
+
+    return process.exit(0);
```

Comment:
Technically this last `process.exit(0)` can happen without throwing the error messages below. If the response's payload is json but doesn't include `data.__typename`, it will just quit the execution without showing the error logs in the below catch method. Maybe instead if the `__typename` is not found, we throw an error, forcing it to be caught in the block below. For example:

```suggestion
    if (!json.data.__typename) {
      throw new Error(`GraphQL request for didn't include "json.data.__typename"`)
    }
```

Additionally, I'm wondering if we need a log to call out that the WPGraphQL endpoint was found. I think by default we should assume it is. Maybe if anything we include this as a debug message with the endpoint included for debug purposes.